File: URI online/2129.cpp
#include  <bits/stdc++.h>
using namespace std;

int fat[11]={1,1,2,6,4,2,2,4,2,8,8};
long long fatorial(int n){
	if(n<=10) return fat[n];
	return ((n/10)%2==1?4:6)*fatorial(n/5)*fatorial(n%10);
}

main(){

	int n, count = 1;
	
	while(cin >> n){
		cout << "Instancia " << count++ << endl << fatorial(n)%10 << "\n\n";
	}
}

File: URI online/2129.py
count = 0
def fatorial(n):
	final = 1
	for i in range(1,n+1):
		final = final * i 
		while(final%10==0):
			final = int(final /10)
	return final%10
		

def saida(n):
	final = fatorial(n)
	if count != 1:
                print()
	print("Instancia",count)
	print(final)

while True:
	try:
		fat = int(input())
		count= count+1
		saida(fat)
	except:
		break
